3rdparty/halrvv/rvv071/halrvv071.hpp

```diff
@@ -0,0 +1,149 @@
+ *                        (3-clause BSD License)
+ *
+ * Copyright (C) 2016, NVIDIA Corporation, all rights reserved.
+ * Third party copyrights are property of their respective owners.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ *   * Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *
+ *   * Redistributions in binary form must reproduce the above copyright notice,
+ *     this list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *
+ *   * Neither the names of the copyright holders nor the names of the contributors
+ *     may be used to endorse or promote products derived from this software
+ *     without specific prior written permission.
+ *
+ * This software is provided by the copyright holders and contributors "as is" and
+ * any express or implied warranties, including, but not limited to, the implied
+ * warranties of merchantability and fitness for a particular purpose are disclaimed.
+ * In no event shall copyright holders or contributors be liable for any direct,
+ * indirect, incidental, special, exemplary, or consequential damages
+ * (including, but not limited to, procurement of substitute goods or services;
+ * loss of use, data, or profits; or business interruption) however caused
+ * and on any theory of liability, whether in contract, strict liability,
+ * or tort (including negligence or otherwise) arising in any way out of
+ * the use of this software, even if advised of the possibility of such damage.
+ */
+
+#ifndef OPENCV_HALRVV071_HAL_HPP_INCLUDED
+#define OPENCV_HALRVV071_HAL_HPP_INCLUDED
+
+#if __GNUC__ > 8 || \
+    (__GNUC__ == 8 && (__GNUC_MINOR__ > 4 || \
+                       (__GNUC_MINOR__ == 4 && \
+                        __GNUC_PATCHLEVEL__ > 0)))
+
+#ifndef CV_HAL_RVV_071_ENABLED
+#  if defined(__THEAD_VERSION__) && defined(__riscv_v) && __riscv_v == 7000
+#    define CV_HAL_RVV_071_ENABLED 1
+#  else
+#    define CV_HAL_RVV_071_ENABLED 0
+#  endif
+#endif
+
+#endif /* GCC version check*/
+
+#if CV_HAL_RVV_071_ENABLED
+
+#include "opencv2/core/hal/interface.h"
+
+#include <riscv_vector.h>
+
+#undef cv_hal_cvtBGRtoBGR
+#define cv_hal_cvtBGRtoBGR cvt_hal_BGRtoBGR
+
+static unsigned char index_array_32 [32] 
+                        { 2, 1, 0, 3, 6, 5, 4, 7, 10, 9, 8, 11, 14, 13, 12, 15, 18, 17, 16, 19, 22, 21, 20, 23, 26, 25, 24, 27, 30, 29, 28, 31  };
+
+static unsigned char index_array_24 [24]
+                        { 2, 1, 0, 5, 4, 3, 8, 7, 6, 11, 10, 9, 14, 13, 12, 17, 16, 15, 20, 19, 18, 23, 22, 21  };
+
+static void cvt_vector(const uchar* src, uchar * dst, uchar * index, int n, int scn, int dcn, int bi, int vsize_pixels, int vsize)
+{
+    vuint8m2_t vec_index = vle8_v_u8m2(index, vsize);
```

Comment:
Let's use hardcoded constant instead of `vsize`, because we assume `-mrvv-vector-bits=128` compile parameter.

As I understand, there will be two variants of this function.

Reply:
If you intend to use mrvv-vector-bits=128 compile parameter, then in addition to using a constant value, you need to use LMUL=1. Am I understanding this correctly?

Reply:
> As I understand, there will be two variants of this function.
You can limit yourself to one implementation, because the definition of a constant will be through a condition, based on the value of scn, we can pass the literal to the constant argument of the function